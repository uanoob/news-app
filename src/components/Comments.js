import React, { Component } from 'react';
import PropTypes from 'prop-types';

import ToggleOpen from '../decorators/toggleOpen';
import Comment from '../containers/Comment';

class Comments extends Component {
  static defaultProps = {
    comments: [],
  };

  static propTypes = {
    comments: PropTypes.arrayOf(
      PropTypes.shape({
        date: PropTypes.string.isRequired,
        text: PropTypes.string.isRequired,
        user: PropTypes.string.isRequired,
        _id: PropTypes.string.isRequired,
      }),
    ),
    isOpen: PropTypes.bool.isRequired,
    toggleOpen: PropTypes.func.isRequired,
  };

  getCommentsElement = () => {
    const { comments } = this.props;
    return comments.length !== 0 ? (
      comments.map(comment => (
        <li key={comment._id}>
          <Comment comment={comment} />
        </li>
      ))
    ) : (
      <div>No comments yep</div>
    );
  };

  render() {
    const { isOpen, toggleOpen } = this.props;
    return (
      <div>
        <button type="button" onClick={toggleOpen}>
          {isOpen ? 'Hide comments' : 'Show comments'}
        </button>
        {isOpen ? <ul>{this.getCommentsElement()}</ul> : null}
      </div>
    );
  }
}

export default ToggleOpen(Comments);
