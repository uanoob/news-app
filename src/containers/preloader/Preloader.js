import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';

const styles = () => ({
  progress: {
    marginTop: 50,
  },
});

function CircularIndeterminate(props) {
  const { classes } = props;
  return (
    <div>
      <CircularProgress
        className={classes.progress}
        color="primary"
        size={100}
      />
    </div>
  );
}

CircularIndeterminate.propTypes = {
  classes: PropTypes.shape({
    progress: PropTypes.string.isRequired,
  }).isRequired,
};

export default withStyles(styles)(CircularIndeterminate);
