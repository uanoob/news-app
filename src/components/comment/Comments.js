import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import { getAllCommentsByArticleId, clearComments } from '../../store/actions';
import Preloader from '../../containers/preloader/Preloader';
import Comment from './Comment';

const styles = theme => ({
  root: {
    width: '100%',
    backgroundColor: theme.palette.background.paper,
    textAlign: 'center',
  },
});

export class CommentsComponent extends Component {
  state = {};

  componentDidMount() {
    const { onGetAllCommentsByArticleId, articleId } = this.props;
    onGetAllCommentsByArticleId(articleId);
  }

  componentWillUnmount() {
    const { onClearComments } = this.props;
    onClearComments();
  }

  render() {
    const {
      classes, comments, isLoading, isLoaded, errorMsg,
    } = this.props;
    return (
      <Fragment>
        {comments && comments.length !== 0 && (
          <div className={classes.root}>
            <List dense>
              {comments.map(comment => (
                <ListItem key={comment._id}>
                  <Comment comment={comment} />
                </ListItem>
              ))}
            </List>
          </div>
        )}
        {isLoading && !isLoaded ? <Preloader /> : null}
        {isLoaded && !comments.length ? (
          <span id="comments-empty-message">No comments yep</span>
        ) : null}
        {errorMsg && <span id="comments-error-message">{errorMsg}</span>}
      </Fragment>
    );
  }
}

CommentsComponent.defaultProps = {
  comments: [],
  errorMsg: null,
};

CommentsComponent.propTypes = {
  classes: PropTypes.shape({
    root: PropTypes.string.isRequired,
  }).isRequired,
  comments: PropTypes.arrayOf(
    PropTypes.shape({
      _id: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired,
      author_id: PropTypes.string.isRequired,
      author_name: PropTypes.string.isRequired,
      article_id: PropTypes.string.isRequired,
      created_at: PropTypes.string.isRequired,
      updated_at: PropTypes.string.isRequired,
    }),
  ),
  onGetAllCommentsByArticleId: PropTypes.func.isRequired,
  onClearComments: PropTypes.func.isRequired,
  articleId: PropTypes.string.isRequired,
  isLoading: PropTypes.bool.isRequired,
  isLoaded: PropTypes.bool.isRequired,
  errorMsg: PropTypes.string,
};

const mapStateToProps = state => ({
  comments: state.comments.comments,
  isLoading: state.comments.loading,
  isLoaded: state.comments.loaded,
});

const mapDispatchToProps = {
  onGetAllCommentsByArticleId: getAllCommentsByArticleId,
  onClearComments: clearComments,
};

const Comments = connect(
  mapStateToProps,
  mapDispatchToProps,
)(withStyles(styles)(CommentsComponent));

export default Comments;
