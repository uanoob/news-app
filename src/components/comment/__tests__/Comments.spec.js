import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import List from '@material-ui/core/List';
import Comments, { CommentsComponent } from '../Comments';
import Preloader from '../../../containers/preloader/Preloader';

describe('Comments', () => {
  const props = {
    classes: { root: '' },
    comments: [],
    articleId: '',
    isLoading: false,
    isLoaded: false,
    errorMsg: null,
    onGetAllCommentsByArticleId: () => {},
    onClearComments: () => {},
  };

  describe('CommentsComponent initial', () => {
    const mockGetAllCommentsByArticleId = jest.fn();
    const nextProps = {
      ...props,
      onGetAllCommentsByArticleId: mockGetAllCommentsByArticleId,
    };
    const wrapper = shallow(<CommentsComponent {...nextProps} />);

    it('render correctly CommentsComponent', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('onGetAllCommentsByArticleId method should have been called', () => {
      expect(mockGetAllCommentsByArticleId).toBeCalledTimes(1);
    });
    it('not render <List />', () => {
      expect(wrapper.find(List)).toHaveLength(0);
    });
    it('not render <Preloader />', () => {
      expect(wrapper.find(Preloader)).toHaveLength(0);
    });
    it('not render errorMsg', () => {
      expect(wrapper.find('#comments-error-message')).toHaveLength(0);
    });
  });

  describe('CommentsComponent isLoading', () => {
    const nextProps = {
      ...props,
      isLoading: true,
    };
    const wrapper = shallow(<CommentsComponent {...nextProps} />);

    it('render correctly CommentsComponent', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('render Preloader', () => {
      expect(wrapper.find(Preloader)).toHaveLength(1);
    });
    it('not render <List />', () => {
      expect(wrapper.find(List)).toHaveLength(0);
    });
    it('not render errorMsg', () => {
      expect(wrapper.find('#comments-error-message')).toHaveLength(0);
    });
  });

  describe('CommentsComponent Loaded & comments array empty', () => {
    const nextProps = {
      ...props,
      isLoading: false,
      isLoaded: true,
    };
    const wrapper = shallow(<CommentsComponent {...nextProps} />);

    it('render correctly CommentsComponent', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('not render Preloader', () => {
      expect(wrapper.find(Preloader)).toHaveLength(0);
    });
    it('not render <List />', () => {
      expect(wrapper.find(List)).toHaveLength(0);
    });
    it('not render errorMsg', () => {
      expect(wrapper.find('#comments-comments-error-message')).toHaveLength(0);
    });
    it('render No comments yep', () => {
      expect(wrapper.find('#comments-empty-message')).toHaveLength(1);
    });
  });

  describe('CommentsComponent render <List />', () => {
    const nextProps = {
      ...props,
      comments: [
        {
          _id: 'PropTypes.string.isRequired',
          text: 'PropTypes.string.isRequired',
          author_id: 'PropTypes.string.isRequired',
          author_name: 'PropTypes.string.isRequired',
          article_id: 'PropTypes.string.isRequired',
          created_at: 'PropTypes.string.isRequired',
          updated_at: 'PropTypes.string.isRequired',
        },
      ],
      isLoading: false,
      isLoaded: true,
    };
    const wrapper = shallow(<CommentsComponent {...nextProps} />);

    it('render correctly CommentsComponent', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('renders <List /> template', () => {
      expect(wrapper.find(List)).toHaveLength(1);
    });
    it('not render <Preloader />', () => {
      expect(wrapper.find(Preloader)).toHaveLength(0);
    });
    it('not render errorMsg', () => {
      expect(wrapper.find('#comments-error-message')).toHaveLength(0);
    });
  });

  describe('CommentsComponent with errorMsg', () => {
    const nextProps = {
      ...props,
      errorMsg: 'Something going wrong',
    };
    const wrapper = shallow(<CommentsComponent {...nextProps} />);

    it('render correctly CommentsComponent', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('renders errorMsg', () => {
      expect(wrapper.find('#comments-error-message').text()).toEqual(
        nextProps.errorMsg,
      );
    });
  });

  describe('CommentsComponent Will Unmount', () => {
    const mockClearComments = jest.fn();
    const nextProps = {
      ...props,
      onClearComments: mockClearComments,
    };
    const wrapper = shallow(<CommentsComponent {...nextProps} />);
    it('render correctly CommentsComponent', () => {
      expect(wrapper).toMatchSnapshot();
    });
    wrapper.unmount();
    it('clearComments method should have been called', () => {
      expect(mockClearComments).toBeCalledTimes(1);
    });
  });

  describe('Comments: mapStateToProps & mapDispatchToProps', () => {
    const mockStore = configureMockStore();
    const getAllCommentsByArticleId = () => ({
      type: 'GET_ALL_COMMENTS_BY_ARTICLE_ID',
    });
    const clearComments = () => ({
      type: 'CLEAR_COMMENTS',
    });
    const comments = [
      {
        _id: 'PropTypes.string.isRequired',
        text: 'PropTypes.string.isRequired',
        author_id: 'PropTypes.string.isRequired',
        author_name: 'PropTypes.string.isRequired',
        article_id: 'PropTypes.string.isRequired',
        created_at: 'PropTypes.string.isRequired',
        updated_at: 'PropTypes.string.isRequired',
      },
    ];
    const initialState = {
      comments,
      isLoading: false,
      isLoaded: false,
    };
    let store;
    let wrapper;
    beforeEach(() => {
      store = mockStore(initialState);
      wrapper = shallow(<Comments store={store} />);
    });
    it('the state values were correctly passed as props', () => {
      expect(wrapper.props().store.getState()).toEqual(initialState);
    });
    it('the store dispatched action: GET_ALL_COMMENTS_BY_ARTICLE_ID', () => {
      store.dispatch(getAllCommentsByArticleId());
      const actions = store.getActions();
      const expectedPayload = { type: 'GET_ALL_COMMENTS_BY_ARTICLE_ID' };
      expect(actions).toEqual([expectedPayload]);
    });
    it('the store dispatched the action: CLEAR_COMMENTS', () => {
      store.dispatch(clearComments());
      const actions = store.getActions();
      const expectedPayload = { type: 'CLEAR_COMMENTS' };
      expect(actions).toEqual([expectedPayload]);
    });
  });
});
