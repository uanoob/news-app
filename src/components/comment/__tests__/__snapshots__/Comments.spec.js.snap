// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments CommentsComponent Loaded & comments array empty render correctly CommentsComponent 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentsComponent
    articleId=""
    classes={
      Object {
        "root": "",
      }
    }
    comments={Array []}
    errorMsg={null}
    isLoaded={true}
    isLoading={false}
    onClearComments={[Function]}
    onGetAllCommentsByArticleId={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        false,
        null,
        <span
          id="comments-empty-message"
        >
          No comments yep
        </span>,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No comments yep",
          "id": "comments-empty-message",
        },
        "ref": null,
        "rendered": "No comments yep",
        "type": "span",
      },
      null,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          false,
          null,
          <span
            id="comments-empty-message"
          >
            No comments yep
          </span>,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No comments yep",
            "id": "comments-empty-message",
          },
          "ref": null,
          "rendered": "No comments yep",
          "type": "span",
        },
        null,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Comments CommentsComponent Will Unmount render correctly CommentsComponent 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentsComponent
    articleId=""
    classes={
      Object {
        "root": "",
      }
    }
    comments={Array []}
    errorMsg={null}
    isLoaded={false}
    isLoading={false}
    onClearComments={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onGetAllCommentsByArticleId={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        false,
        null,
        null,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      null,
      null,
      null,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          false,
          null,
          null,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        null,
        null,
        null,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Comments CommentsComponent initial render correctly CommentsComponent 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentsComponent
    articleId=""
    classes={
      Object {
        "root": "",
      }
    }
    comments={Array []}
    errorMsg={null}
    isLoaded={false}
    isLoading={false}
    onClearComments={[Function]}
    onGetAllCommentsByArticleId={
      [MockFunction] {
        "calls": Array [
          Array [
            "",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        false,
        null,
        null,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      null,
      null,
      null,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          false,
          null,
          null,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        null,
        null,
        null,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Comments CommentsComponent isLoading render correctly CommentsComponent 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentsComponent
    articleId=""
    classes={
      Object {
        "root": "",
      }
    }
    comments={Array []}
    errorMsg={null}
    isLoaded={false}
    isLoading={true}
    onClearComments={[Function]}
    onGetAllCommentsByArticleId={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        false,
        <WithStyles(CircularIndeterminate) />,
        null,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
      null,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          false,
          <WithStyles(CircularIndeterminate) />,
          null,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
        null,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Comments CommentsComponent render <List /> render correctly CommentsComponent 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentsComponent
    articleId=""
    classes={
      Object {
        "root": "",
      }
    }
    comments={
      Array [
        Object {
          "_id": "PropTypes.string.isRequired",
          "article_id": "PropTypes.string.isRequired",
          "author_id": "PropTypes.string.isRequired",
          "author_name": "PropTypes.string.isRequired",
          "created_at": "PropTypes.string.isRequired",
          "text": "PropTypes.string.isRequired",
          "updated_at": "PropTypes.string.isRequired",
        },
      ]
    }
    errorMsg={null}
    isLoaded={true}
    isLoading={false}
    onClearComments={[Function]}
    onGetAllCommentsByArticleId={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className=""
        >
          <WithStyles(List)
            dense={true}
          >
            <WithStyles(ListItem)>
              <Connect(WithStyles(Comment))
                comment={
                  Object {
                    "_id": "PropTypes.string.isRequired",
                    "article_id": "PropTypes.string.isRequired",
                    "author_id": "PropTypes.string.isRequired",
                    "author_name": "PropTypes.string.isRequired",
                    "created_at": "PropTypes.string.isRequired",
                    "text": "PropTypes.string.isRequired",
                    "updated_at": "PropTypes.string.isRequired",
                  }
                }
              />
            </WithStyles(ListItem)>
          </WithStyles(List)>
        </div>,
        null,
        null,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <WithStyles(List)
            dense={true}
          >
            <WithStyles(ListItem)>
              <Connect(WithStyles(Comment))
                comment={
                  Object {
                    "_id": "PropTypes.string.isRequired",
                    "article_id": "PropTypes.string.isRequired",
                    "author_id": "PropTypes.string.isRequired",
                    "author_name": "PropTypes.string.isRequired",
                    "created_at": "PropTypes.string.isRequired",
                    "text": "PropTypes.string.isRequired",
                    "updated_at": "PropTypes.string.isRequired",
                  }
                }
              />
            </WithStyles(ListItem)>
          </WithStyles(List)>,
          "className": "",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(ListItem)>
                <Connect(WithStyles(Comment))
                  comment={
                    Object {
                      "_id": "PropTypes.string.isRequired",
                      "article_id": "PropTypes.string.isRequired",
                      "author_id": "PropTypes.string.isRequired",
                      "author_name": "PropTypes.string.isRequired",
                      "created_at": "PropTypes.string.isRequired",
                      "text": "PropTypes.string.isRequired",
                      "updated_at": "PropTypes.string.isRequired",
                    }
                  }
                />
              </WithStyles(ListItem)>,
            ],
            "dense": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "PropTypes.string.isRequired",
              "nodeType": "class",
              "props": Object {
                "children": <Connect(WithStyles(Comment))
                  comment={
                    Object {
                      "_id": "PropTypes.string.isRequired",
                      "article_id": "PropTypes.string.isRequired",
                      "author_id": "PropTypes.string.isRequired",
                      "author_name": "PropTypes.string.isRequired",
                      "created_at": "PropTypes.string.isRequired",
                      "text": "PropTypes.string.isRequired",
                      "updated_at": "PropTypes.string.isRequired",
                    }
                  }
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "comment": Object {
                    "_id": "PropTypes.string.isRequired",
                    "article_id": "PropTypes.string.isRequired",
                    "author_id": "PropTypes.string.isRequired",
                    "author_name": "PropTypes.string.isRequired",
                    "created_at": "PropTypes.string.isRequired",
                    "text": "PropTypes.string.isRequired",
                    "updated_at": "PropTypes.string.isRequired",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      null,
      null,
      null,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className=""
          >
            <WithStyles(List)
              dense={true}
            >
              <WithStyles(ListItem)>
                <Connect(WithStyles(Comment))
                  comment={
                    Object {
                      "_id": "PropTypes.string.isRequired",
                      "article_id": "PropTypes.string.isRequired",
                      "author_id": "PropTypes.string.isRequired",
                      "author_name": "PropTypes.string.isRequired",
                      "created_at": "PropTypes.string.isRequired",
                      "text": "PropTypes.string.isRequired",
                      "updated_at": "PropTypes.string.isRequired",
                    }
                  }
                />
              </WithStyles(ListItem)>
            </WithStyles(List)>
          </div>,
          null,
          null,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(List)
              dense={true}
            >
              <WithStyles(ListItem)>
                <Connect(WithStyles(Comment))
                  comment={
                    Object {
                      "_id": "PropTypes.string.isRequired",
                      "article_id": "PropTypes.string.isRequired",
                      "author_id": "PropTypes.string.isRequired",
                      "author_name": "PropTypes.string.isRequired",
                      "created_at": "PropTypes.string.isRequired",
                      "text": "PropTypes.string.isRequired",
                      "updated_at": "PropTypes.string.isRequired",
                    }
                  }
                />
              </WithStyles(ListItem)>
            </WithStyles(List)>,
            "className": "",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(ListItem)>
                  <Connect(WithStyles(Comment))
                    comment={
                      Object {
                        "_id": "PropTypes.string.isRequired",
                        "article_id": "PropTypes.string.isRequired",
                        "author_id": "PropTypes.string.isRequired",
                        "author_name": "PropTypes.string.isRequired",
                        "created_at": "PropTypes.string.isRequired",
                        "text": "PropTypes.string.isRequired",
                        "updated_at": "PropTypes.string.isRequired",
                      }
                    }
                  />
                </WithStyles(ListItem)>,
              ],
              "dense": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "PropTypes.string.isRequired",
                "nodeType": "class",
                "props": Object {
                  "children": <Connect(WithStyles(Comment))
                    comment={
                      Object {
                        "_id": "PropTypes.string.isRequired",
                        "article_id": "PropTypes.string.isRequired",
                        "author_id": "PropTypes.string.isRequired",
                        "author_name": "PropTypes.string.isRequired",
                        "created_at": "PropTypes.string.isRequired",
                        "text": "PropTypes.string.isRequired",
                        "updated_at": "PropTypes.string.isRequired",
                      }
                    }
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "comment": Object {
                      "_id": "PropTypes.string.isRequired",
                      "article_id": "PropTypes.string.isRequired",
                      "author_id": "PropTypes.string.isRequired",
                      "author_name": "PropTypes.string.isRequired",
                      "created_at": "PropTypes.string.isRequired",
                      "text": "PropTypes.string.isRequired",
                      "updated_at": "PropTypes.string.isRequired",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        null,
        null,
        null,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Comments CommentsComponent with errorMsg render correctly CommentsComponent 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentsComponent
    articleId=""
    classes={
      Object {
        "root": "",
      }
    }
    comments={Array []}
    errorMsg="Something going wrong"
    isLoaded={false}
    isLoading={false}
    onClearComments={[Function]}
    onGetAllCommentsByArticleId={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        false,
        null,
        null,
        <span
          id="comments-error-message"
        >
          Something going wrong
        </span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      null,
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Something going wrong",
          "id": "comments-error-message",
        },
        "ref": null,
        "rendered": "Something going wrong",
        "type": "span",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          false,
          null,
          null,
          <span
            id="comments-error-message"
          >
            Something going wrong
          </span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        null,
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Something going wrong",
            "id": "comments-error-message",
          },
          "ref": null,
          "rendered": "Something going wrong",
          "type": "span",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
